### PR TITLE
feat: Convert docker registries to connectors

### PR DESCRIPTION
**Pull request by naphta**
Went on a little bit of a mission with this PR, but ultimately there were two issues encountered when trying to use the chart and these are my solutions.

Firstly the chart had a notion of using docker registries and having ingress to them, that's great but inside nexus they're called connectors, and there's nothing to say they need to be locked down to docker, simply that they're http. My use case was docker and nuget, as an example.

Secondly since I'm using Istio ingress the lack of good configuration of the pathType, and being able to set the ingressClass outside of an annotation was causing me some issues so I modernised the ingress manifests to the latest (but backwards compatible) versions with some sensible pathType's of `Prefix` set.

Happy to make any changes desired to make this mergeable since I acknowledge I've changed a fair bit of the interface for users of the chart by tweaking the ingress object.

### CODE REVIEW

**Comment by naphta** [comment]
@jflinchbaugh Just wondering if you can review this. (I had a nose through other PR's and you've some approving before)

**Comment by ruckc** [comment]
The commit looks like it's doing alot more than just the ingress/connector changes... why all the reworking of labels and all the other toYaml changes?  I can't easily separate the "yaml cleanup" stuff from what you are actually asking about.

**Comment by naphta** [comment]
> The commit looks like it's doing alot more than just the ingress/connector changes... why all the reworking of labels and all the other toYaml changes? I can't easily separate the "yaml cleanup" stuff from what you are actually asking about.

I made some of those changes because testing locally some of the logic around values which werent defined in the values.yaml tricky. The labels change I made simply because it was shorter and easier to understand when writing the part into the ingress resource and I decided to make that change everywhere rather than just in the ingress.

Happy to split up the stylistic changes and the ingress changes into two separate commits either in this PR or into two different PR's, whichever is easiest.

**Comment by ruckc** [comment]
while i'm not with sonatype, i'd prefer stylistic stuff in a separate commit from logic changes.

separately, i've just switched myself to using separate ingresses for all of my repositories to map them to subdomains instead of the path'd syntax.  The issue with doing that in the chart that i see is it is very ingress controller specific (i.e. Nginx requires rewrite-target).

**Comment by naphta** [comment]
I've done a bit of clean-up (finally), and split up the PR into two distinct commits with differing commit messages to describe what they're doing. It would have been a bit messy to split it up even further but does this help @ruckc?

File: charts/nexus-repository-manager/templates/ingress.yaml [hunk 1]
```diff
@@ -1,73 +1,74 @@
...
-    {{- if $.Values.nexus.extraLabels }}
-      {{- with $.Values.nexus.extraLabels }}
-        {{ toYaml . | indent 4 }}
-      {{- end }}
-    {{- end }}
-  {{- with $.Values.ingress.annotations }}
-  annotations:
-    {{- toYaml . | nindent 4 }}
-  {{- end }}
+  name: {{ $fullName | trunc 49 }}-docker-{{ $connector.port }}
```
Comment (ruckc) [inline charts/nexus-repository-manager/templates/ingress.yaml:47]: probably should change docker to connector?  and 49 to a smaller number so the name doesn't hit 63

Comment (naphta) [reply charts/nexus-repository-manager/templates/ingress.yaml]: Good spot, I've just pushed a commit which fixes this. 

I calculated number (roughly) with
```63 - len("-connector-65535")```

**Comment by naphta** [comment]
@ruckc Was there anything else you saw worth discussing / changing? Keen to get this PR merged!

**Comment by ruckc** [comment]
i didn't see anything else... though i've been playing with ingress's directly to the default port to the full repository url without having to add extra ports... and that works well, but it requires custom annotations depending on ingress controller, which i think is outside the scope of a general purpose helm chart

**Comment by jflinchbaugh** [comment]
sorry we left this get too old and accumulate too many conflicts.

**Pull request closed without merging**